### PR TITLE
Refactor code

### LINKED ISSUE #57: Additionnal refactoring
Writing down a few problems I've seen in the code while reviewing.

- do another pass of variables names fix (`referer` ? )
- remove variables like `c`
- use keyword arguments wherever function calls are not self-explainatory `("", c, scraper)`
- functions that expects the scraper as parameter are odd (dl_dependencies).
- function names too generic (`Connection`)

### PR DESCRIPTION
**Pull request by satyamtg**
This would fix #57, #52, #56.Also addresses #48

This one does several changes as described below (I've tried to keep the git history clean) -
- The connection.py is refactored. Also variables like c, connection are removed and that's replaced with instance_connection. Also the connection.py module is renamed to instance_connection.py. This module has the following changes - 
    - the configs are renamed and moved to INSTANCE_CONFIGS in constants.py
    - variable names are now clearer
- Refactored utils.py -
    - Unused imports are removed.
    - dl_dependencies moved to scraper.py
    - get_filetype is removed as it wasn't used anymore
    - several functions are refactored to use clearer names and shorter code
- Refactored scraper.py -
    - added dl_dependencies
    - exploded dl_dependencies into several parts (mainly functions named get_<tag>_from_html())
    - small fixes here and there 
- The issue #52 is now fixed by properly creating objects of Unavailable type for unsupported/unofflineable xblocks. A dictionary containing such xblocks and their respective error messages are put in constants.py.
- The issue #56 is fixed in drag_and_drop_v2.py. Apparently that was a beautifulsoup error
- Some long functions are refactored, so this addresses #48 but there's still some scope for this in annex.py. I have done some changes there in my local version, but what's here right now is pretty consistent and that's why I think it may be easier to do annex.py refactor separately. So will keep #48 open for now.
- Some links were also downloaded while downloading docs in dl_dependencies. So, dropped downloading relative links for now. Will fix this with #24 
- Also renamed xblock_extractor files to have their names in lowercase to match the naming scheme in the repo
- Used correct outtmpl in youtube_dl and fixed youtube video downloading

### CODE REVIEW

File: openedx2zim/xblocks_extractor/problem.py [hunk 1]
```diff
@@ -72,11 +80,11 @@ def download(self, c):
             )
             if "success" in answers_content:
                 """
-                    #IMPROUVEMENT connection , same as hint ?
+                    #IMPROUVEMENT instance_connection , same as hint ?
```
Comment (rgaudin) [inline openedx2zim/xblocks_extractor/problem.py:83]: fix the typo in comment maybe?

File: openedx2zim/constants.py [hunk 2]
```diff
@@ -28,6 +28,45 @@
...
+    "default": {
+        "login_page": "/login_ajax",
+        "account_page": "/account/settings",
+        "course_page_name": "/info",
+        "course_prefix": "/courses/",
+        "instance_url": "",
+    },
+}
+
+UNSUPPORTED_XBLOCKS = {"grademebutton": "Grade Me (Unavialable Offline)"}
```
[the remark this answers is not in the document]
Comment (satyamtg) [reply openedx2zim/constants.py:75]: Fixed

File: openedx2zim/xblocks_extractor/problem.py (hunk 1 above)
Comment (satyamtg) [reply openedx2zim/xblocks_extractor/problem.py:83]: Fixed the typo. But kept the comment as that'll allow us to look into #27 later on

File: openedx2zim/constants.py [hunk 3]
```diff
@@ -28,6 +28,45 @@
     "gif": "v1",
 }
 
+DOWNLOADABLE_CONTENT = [
```
[the remark this answers is not in the document]
Comment (satyamtg) [reply openedx2zim/constants.py:31]: Changed to DOWNLOADABLE_EXTENSIONS

**Pull request merged**